src/win/win_impl.rs

```diff
@@ -151,10 +151,20 @@ impl Mouse for Enigo {
 
     fn move_mouse(&mut self, x: i32, y: i32, coordinate: Coordinate) -> InputResult<()> {
         debug!("\x1b[93mmove_mouse(x: {x:?}, y: {y:?}, coordinate:{coordinate:?})\x1b[0m");
-        let flags = if coordinate == Coordinate::Abs {
-            MOUSEEVENTF_MOVE | MOUSEEVENTF_ABSOLUTE
+        let (flags, x, y) = if coordinate == Coordinate::Abs {
+            let (w, h) = self.main_display()?;
+            let w = w as i64;
+            let h = h as i64;
+            let x = x as i64;
+            let y = y as i64;
+            // Add w/2 or h/2 to round off
+            // Multiply by 65535 because MOUSEEVENTF_ABSOLUTE flag is set
+            // See https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-mouse_event#remarks
+            let x = (x * 65535 + w / 2 * x.signum()) / w;
+            let y = (y * 65535 + h / 2 * y.signum()) / h;
```

Comment:
The current code is off by one for some pixels, on a 1080p monitor 512,512 results in a cursor position of 511,511 for example. For a width of 1080, the highest reachable pixel is 1079, so 65535 maps to 1079.

I think these two lines should be:
```rust
let y = (65535.0 / (h - 1) as f64) * (y as f64);
let x = (65535.0 / (w - 1) as f64) * (x as f64);
```

This change seems to work for the coordinates I tested; two 1080p monitors side by side, right being primary. Both negative and positive x coordinates exactly achieve the desired value.

Reply:
Feel free to cherry-pick 6dec1e29b355d7df1f3d4821b90e2204aa5273df which contains this change as well as a comment explaining it.

Reply:
fixed